errors/errors.go

```diff
@@ -15,209 +15,310 @@ package errors
+			case *errdetails.RequestInfo:
+				desc := typedDetail.ProtoReflect().Descriptor()
+				detailMap := map[string]interface{}{
+					"@type":        typeGoogleAPI + desc.FullName(),
+					"request_id":   typedDetail.RequestId,
+					"serving_data": typedDetail.ServingData,
+				}
+				errJson["details"].([]interface{})[i] = detailMap
+			case *errdetails.ResourceInfo:
+				desc := typedDetail.ProtoReflect().Descriptor()
+				detailMap := map[string]interface{}{
+					"@type":         typeGoogleAPI + desc.FullName(),
+					"resource_type": typedDetail.ResourceType,
+					"resource_name": typedDetail.ResourceName,
+					"owner":         typedDetail.Owner,
+					"description":   typedDetail.Description,
+				}
+				errJson["details"].([]interface{})[i] = detailMap
+			case *errdetails.Help:
+				desc := typedDetail.ProtoReflect().Descriptor()
+				detailMap := map[string]interface{}{
+					"@type": typeGoogleAPI + desc.FullName(),
+					"links": typedDetail.Links,
+				}
+				errJson["details"].([]interface{})[i] = detailMap
+			case *errdetails.LocalizedMessage:
+				desc := typedDetail.ProtoReflect().Descriptor()
+				detailMap := map[string]interface{}{
+					"@type":   typeGoogleAPI + desc.FullName(),
+					"locale":  typedDetail.Locale,
+					"message": typedDetail.Message,
+				}
+				errJson["details"].([]interface{})[i] = detailMap
+			case *errdetails.QuotaFailure_Violation:
+				desc := typedDetail.ProtoReflect().Descriptor()
+				detailMap := map[string]interface{}{
+					"@type":       typeGoogleAPI + desc.FullName(),
+					"subject":     typedDetail.Subject,
+					"description": typedDetail.Description,
+				}
+				errJson["details"].([]interface{})[i] = detailMap
+			case *errdetails.PreconditionFailure_Violation:
+				desc := typedDetail.ProtoReflect().Descriptor()
+				detailMap := map[string]interface{}{
+					"@type":       typeGoogleAPI + desc.FullName(),
+					"subject":     typedDetail.Subject,
+					"description": typedDetail.Description,
+					"type":        typedDetail.Type,
+				}
+				errJson["details"].([]interface{})[i] = detailMap
+			case *errdetails.BadRequest_FieldViolation:
+				desc := typedDetail.ProtoReflect().Descriptor()
+				detailMap := map[string]interface{}{
+					"@type":       typeGoogleAPI + desc.FullName(),
+					"field":       typedDetail.Field,
+					"description": typedDetail.Description,
+				}
+				errJson["details"].([]interface{})[i] = detailMap
+			case *errdetails.Help_Link:
+				desc := typedDetail.ProtoReflect().Descriptor()
+				detailMap := map[string]interface{}{
+					"@type":       typeGoogleAPI + desc.FullName(),
+					"description": typedDetail.Description,
+					"url":         typedDetail.Url,
+				}
+				errJson["details"].([]interface{})[i] = detailMap
+			default:
+				log.Debugf("Failed to convert error details due to incorrect type. \nSee types here: https://github.com/googleapis/googleapis/blob/master/google/rpc/error_details.proto. \nDetail: %s", detail)
```

Comment:
Perhaps this should panic as it would indicate a development-time error only?

Reply:
I agree we should catch this in development time, but Im worried that not all http cases get tested and this would come up during http only, not grpc. This is an edge case we shouldn't really get into, but just in case I feel like it's better to still give the details and not panic.